### PR TITLE
bump docker version to 4.5.1

### PR DESCRIPTION
**Pull request by mjeffryes**
Upgrade the pinned docker dependency to 4.5.1.
Fixes: #1219

### CODE REVIEW

**Comment by mjeffryes** [comment]
Waiting on the release of https://github.com/pulumi/pulumi-aws/pull/3426 in pulumi-aws to unblock the tests here.

**Review by iwahbe — APPROVED** [review]

**Pull request merged**
